src/main/java/org/ice4j/ice/harvest/MappingCandidateHarvesters.java

```diff
@@ -283,42 +283,53 @@ public StunMappingCandidateHarvester call()
 
         // Now run discover() on all created harvesters in parallel and pick
         // the ones which succeeded.
-        ExecutorService es = Executors.newFixedThreadPool(tasks.size());
+        ExecutorService es = Executors.newFixedThreadPool(tasks.size(),
+            runnable -> {
+                Thread thread = new Thread(runnable);
+                thread.setName("ICE Harvester Executor");
```

Comment:
You'll have `tasks.size()` threads named `ICE Harvester Executor` with this. The default factory takes care of that: `pool-1-thread-1`. Again, depends on the size of the tasks list if this is relevant.